### PR TITLE
OSSM-6347: SM 3.0 Deploying Istio

### PR DESCRIPTION
**Pull request by rh-tokeefe**
Version(s): 
Service Mesh 3.0

Issue: 
https://issues.redhat.com/browse/OSSM-6374

Link to docs preview:
https://75094--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/install/ossm-installing-openshift-service-mesh.html#about-istio-deployment

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ossm-creating-istio-system-project-using-console.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+.Prerequisties
+
+
+.Procedure
+
+. In the {ocp-product-title} web console, click *Home* -> *Projects*.
+
+. Click *Create Project*.
+
+. At the prompt, you must enter a name for the project in the Name field. For example, `istio-system`. The Operator deploys Istio to the project you specify. The other fields provide supplementary information and are optional.
```
Comment (fjglira) [inline modules/ossm-creating-istio-system-project-using-console.adoc:15]: I think is not needed to add this information: `The Operator deploys Istio to the project you specify. The other fields provide supplementary information and are optional.`. In the next section, we already specify that we are going to create the resource in the namespace created before. Regarding the extra fields when you create the resource maybe is better to say this directly in the steps or in a Note in the section. Something like this:
```
Note: the `istio` resource definition contains supplementary information that is optional to be defined
```

File: modules/ossm-creating-istiocni-resource-using-console.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+. Click *IstioCNI*.
+
+. Click *Create IstioCNI*.
+
+. Ensure that the name is `default`.
+
+. Click *Create*. This action deploys the Istio CNI plugin.
++
+When `State: Healthy` appears in the *Status* column, the Istio CNI plugin is successfully deployed.
```
Comment (fjglira) [inline modules/ossm-creating-istiocni-resource-using-console.adoc:27]: In the document, this text appears misaligned with the rest of the content; it is on the same level as the bullet points. Please take a look in the docs

File: modules/ossm-creating-istio-resource-using-console.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+. Click *Istio*.
+
+. Click *Create Istio*.
+
+. Select the `istio-system` project from the *Namespace* drop-down menu.
+
+. Click *Create*. This action deploys the {istio} control plane.
++
+When `State: Healthy` appears in the *Status* column, {istio} is successfully deployed.
```
Comment (fjglira) [inline modules/ossm-creating-istio-resource-using-console.adoc:27]: Same here

File: modules/ossm-creating-istio-cni-project-using-console.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+.Prerequisties
+
+
+.Procedure
+
+. In the {ocp-product-title} web console, click *Home* -> *Projects*.
+
+. Click *Create Project*.
+
+. At the prompt, you must enter a name for the project in the Name field. For example, `istio-cni`. The Operator deploys {istio} to the project you specify. The other fields provide supplementary information and are optional.
```
Comment (fjglira) [inline modules/ossm-creating-istio-cni-project-using-console.adoc:15]: Here says: `The Operator deploys {istio} to the project you specify`, but the step is to create the `istio-cni`. But, please take a look at the comments that I added to the istio-system project creation. I think this comments should be moved

**Comment by fjglira** [comment]
lgtm, I only have small comments

File: modules/ossm-creating-istiocni-resource-using-console.adoc (hunk 2 above)
Comment (rh-tokeefe) [reply modules/ossm-creating-istiocni-resource-using-console.adoc:27]: That was intentional. It lines up with where the text starts for step 7. It's just describing when things are done.

File: modules/ossm-creating-istio-resource-using-console.adoc (hunk 3 above)
Comment (rh-tokeefe) [reply modules/ossm-creating-istio-resource-using-console.adoc:27]: same

File: modules/ossm-creating-istio-cni-project-using-console.adoc (hunk 4 above)
Comment (rh-tokeefe) [reply modules/ossm-creating-istio-cni-project-using-console.adoc:17]: I fixed things so the attribute resolves now.

I'll run the other comment by the team just to get more opinions.

**Comment by rh-tokeefe** [comment]
@dgn I'm not sure what the prerequisites should be for these tasks. I'm assuming, similar to deploying the control plane in 2.x, it would this:

The Red Hat OpenShift Service Mesh Operator must be installed.

You are logged in to the OpenShift Container Platform web console as cluster-admin.

Is there anything else? I'm not sure if different permissions are needed for creating the project and creating the resource, but I would think not.

Also, for creating the resource, I can add an additional prereq that indicates that the project to which the resource gets deployed must already exist.

Can you think of anything else as far as prereqs?

File: modules/ossm-creating-istio-cni-project-using-console.adoc (hunk 4 above)
Comment (dgn) [reply modules/ossm-creating-istio-cni-project-using-console.adoc:17]: I think we should remove the sentence `The Operator deploys ...`

**Review by dgn — APPROVED** [review]
lgtm, just one comment

**Comment by dgn** [comment]
> @dgn I'm not sure what the prerequisites should be for these tasks. I'm assuming, similar to deploying the control plane in 2.x, it would be similar to this:
> 
>     * The Red Hat OpenShift Service Mesh Operator must be installed.
> 
>     * You are logged in to the OpenShift Container Platform web console as cluster-admin.

Those prerequisites are good!

> 
> Is there anything else? I'm not sure if different permissions are needed for creating the project and creating the resource, but I would think not.
> 
> Also, for creating the resource, I can add an additional prereq that indicates that the project to which the resource gets deployed must already exist.

Don't think that's needed as we explicitly explain how to create the projects

> 
> Can you think of anything else as far as prereqs?

Nope, LGTM

File: modules/ossm-creating-istio-cni-project-using-console.adoc (hunk 4 above)
Comment (rh-tokeefe) [reply modules/ossm-creating-istio-cni-project-using-console.adoc:17]: @fjglira I just re-read this and now I realize what you mean. Yes, I agree. Change was made.

File: modules/ossm-creating-istio-system-project-using-console.adoc (hunk 1 above)
Comment (rh-tokeefe) [reply modules/ossm-creating-istio-system-project-using-console.adoc]: Made minor change here.

**Comment by gabriel-rh** [comment]
> n this example, `istio-system' is the name of the project.

Looks like a mismatch

**Review by gabriel-rh — APPROVED** [review]
LGTM

**Pull request merged**